### PR TITLE
fix: deeplink fix for partner shows and works

### PR DESCRIPTION
**Pull request by rajsam003**
This PR resolves [MOPLAT-486] 

### Description

|Before|

https://user-images.githubusercontent.com/17421923/202220148-e8d0e2a8-face-487d-bc2b-f0f705a887eb.mov

|After|

https://user-images.githubusercontent.com/17421923/202220185-875ac125-ceea-468f-892f-0addb83ffabf.mov

### PR Checklist

- [ ] I tested my changes on **iOS** / **Android**.
- [ ] I added screenshots or videos to illustrate my changes.
- [ ] I added Tests and Stories for my changes.
- [ ] I added an [app state migration].
- [ ] I hid my changes behind a [feature flag].
- [ ] I have prefixed changes that need to be tested during a release QA with **[NEEDS EXTERNAL QA]** on the changelog.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[MOPLAT-486]: https://artsyproduct.atlassian.net/browse/MOPLAT-486?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/Scenes/Partner/Partner.tsx [hunk 1]
```diff
@@ -52,7 +53,7 @@ class Partner extends React.Component<Props> {
           },
           {
             title: "Artworks",
-            initial: true,
```
Comment (rajsam003) [inline src/app/Scenes/Partner/Partner.tsx:55]: Previously, Artworks tab was the initial default tab by which these two routes point to the same tab `/partner/:partnerID` and `/partner/:partnerID/works`. This didn't make sense to me.

File: src/app/navigation/routes.ts [hunk 2]
```diff
@@ -215,7 +215,14 @@ function getDomainMap(): Record<string, RouteMatcher[] | null> {
...
     addRoute("/partner/:partnerID", "Partner"),
-    addRoute("/partner/:partnerID/works", "Partner"),
+    addRoute("/partner/:partnerID/shows", "Partner", (params) => ({
+      ...params,
+      initialTab: "Shows",
+    })),
+    addRoute("/partner/:partnerID/works", "Partner", (params) => ({
+      ...params,
+      initialTab: "Artworks",
+    })),
```
Comment (rajsam003) [inline src/app/navigation/routes.ts:225]: Now with this change
` /partner/:partnerID` - navigates to Partner's Overview tab
` /partner/:partnerID/works` - navigates to Partner's Artworks tab
` /partner/:partnerID/shows` - navigates to Partner's Shows tab

File: src/app/Scenes/Partner/Partner.tsx [hunk 3]
```diff
@@ -52,7 +53,7 @@ class Partner extends React.Component<Props> {
...
+            initial: initialTab === "Artworks",
```
Comment (gkartalis) [inline src/app/Scenes/Partner/Partner.tsx:56]: 👏

File: src/app/navigation/routes.ts (hunk 2 above)
Comment (gkartalis) [reply src/app/navigation/routes.ts:225]: 👌

**Review by gkartalis — APPROVED** [review]
thanks! 🔥

**Review by gkartalis — APPROVED** [review]

**Review by olerichter00 — APPROVED** [review]

**Pull request merged**